features/cellular/framework/API/CellularNetwork.h

```diff
@@ -257,23 +257,35 @@ class CellularNetwork {
 
     /** Set CIoT optimizations.
      *
-     *  @param supported_opt Supported CIoT EPS optimizations.
+     *  @param supported_opt Supported CIoT EPS optimizations
+     *                       (the HW support can be checked with get_ciot_ue_optimization_config).
      *  @param preferred_opt Preferred CIoT EPS optimizations.
+     *  @param network_support_cb This callback will be called when CIoT network optimisation support is known
      *  @return              NSAPI_ERROR_OK on success
      *                       NSAPI_ERROR_DEVICE_ERROR on failure
      */
-    virtual nsapi_error_t set_ciot_optimization_config(Supported_UE_Opt supported_opt,
-                                                       Preferred_UE_Opt preferred_opt) = 0;
+    virtual nsapi_error_t set_ciot_optimization_config(CIoT_Supported_Opt supported_opt,
+            CIoT_Preferred_UE_Opt preferred_opt,
+            Callback<void(CIoT_Supported_Opt)> network_support_cb) = 0;
```

Comment:
@0xc0170: I assume is breaking because of this chage and the one from line 278 bellow, renaming 
get_ciot_optimization_config ->get_ciot_ue_optimization_config